includes/Listeners/WonderCart.php

```diff
@@ -13,32 +13,156 @@ class WonderCart extends Listener {
 	 * @return void
 	 */
 	public function register_hooks() {
-		add_action('rest_after_insert_yith_campaign', array( $this, 'register_campaign' ), 10 );
+		add_action( 'rest_after_insert_yith_campaign', array( $this, 'register_campaign' ), 10 );
+		add_action( 'yith_sales_edit_campaign_event_modal_opened', array( $this, 'create_campaign_modal_open' ), 10, 2 );
+		add_action( 'yith_sales_edit_campaign_event_campaign_selected', array( $this, 'campaign_selected' ), 10, 2 );
+		add_action( 'yith_sales_edit_campaign_event_campaign_abandoned', array( $this, 'campaign_abandoned' ), 10, 2 );
+		add_action( 'woocommerce_checkout_order_processed', array( $this, 'checkout_campaigns_used' ), 10 );
```

Comment:
It seems that this hook may not work the way we think? https://github.com/woocommerce/woocommerce/issues/28113#issuecomment-717479048

Perhaps the other hook mentioned in that comment would be better?

Also, since 10 is the default priority, we can drop that from the action.